src/api/app/models/bs_request.rb

```diff
@@ -736,6 +736,10 @@ def create_new_review(opts)
     )
     return newreview if newreview.valid?
 
+    if (error_text = newreview.errors.select { |error| error.attribute.in?([:user, :group, :project, :package]) }).any?
```

Comment:
This logic should go inside `Review` class, as `BsRequest` [is asking for information](https://thoughtbot.com/blog/tell-dont-ask) that is perfectly known by `Review`

Reply:
I created a method in the Review model and moved the login there. Please take a look again.